lib/postgrestex.ex

```diff
@@ -133,16 +133,28 @@ defmodule Postgrestex do
 
   @spec select(map(), list()) :: map()
   def select(req, columns) do
-    update_headers(req, %{select: Enum.join(columns, ","), method: "GET"})
+    %{req | params: [{:select, Enum.join(columns, ",")} | req.params], method: "GET"}
   end
 
   @doc """
   Insert a row into currently selected table. Does an insert and update if upsert is set to True
   """
   @doc since: "0.1.0"
-  @spec insert(map(), list(), true | false) :: map()
-  def insert(req, json, upsert \\ false) do
+  def insert(req, json, options) when is_list(options),
+    do: insert(req, json, false, options)
+
+  def insert(req, json, upsert) when is_boolean(upsert),
+    do: insert(req, json, upsert, [])
+
+  @spec insert(map(), map(), true | false, keyword()) :: map()
```

Comment:
It looks a lot cleaner to have insert placed as multiple clauses to be matched against.